packages/broker/src/ConfigWizard.ts

```diff
@@ -82,200 +88,181 @@ const DefaultConfig: Config = {
-            wallet = Wallet.createRandom()
-            this.config.ethereumPrivateKey = wallet.privateKey
-            logger.info('This is your generated private key. Please store it in a secure location')
-            logger.alert(this.config.ethereumPrivateKey)
-        } else if (generateOrImport === 'import') {
-            const privateKey = await this.inquirerSinglePrompt({
-                type: 'input',
-                name: 'privateKey',
-                message: "'Please provide the private key to import'",
-            })
-
-            try {
-                wallet = new Wallet(privateKey)
-                this.config.ethereumPrivateKey = wallet.privateKey
-            } catch (privateKeyError) {
-                throw new Error(`Invalid privateKey provided for import: ${privateKey}`)
+            if (portNumber < MIN_PORT_VALUE || portNumber > MAX_PORT_VALUE) {
+                return `Out of range port ${portNumber} provided (valid range ${MIN_PORT_VALUE}-${MAX_PORT_VALUE})`
             }
 
-        } else {
-            throw new Error(`Invalid option ${generateOrImport} provided`)
+            return true
+        },
+        default: defaultPluginPort
+    })
+})
+
+prompts = prompts.concat(pluginSelectorPrompt).concat(pluginPrompts)
+
+export const getConfigFromAnswers = (answers: any): Config => {
+    const config = { ... DEFAULT_CONFIG, plugins: { ... DEFAULT_CONFIG.plugins } }
+
+    const pluginNames = Object.values(PLUGIN_NAMES)
+    pluginNames.forEach((pluginName) => {
+        const defaultPluginPort = PLUGIN_DEFAULT_PORTS[pluginName]
+        if (answers.selectPlugins && answers.selectPlugins.includes(pluginName)){
+            let pluginConfig = {}
+            if (answers[`${pluginName}Port`] !== defaultPluginPort){
+                if (pluginName === PLUGIN_NAMES.PUBLISH_HTTP) {
+                    // the publishHttp plugin is special, it needs to be added to the config after the other plugins
+                    config.httpServer = {
+                        port: answers[`${pluginName}Port`]
+                    }
+                } else {
+                    // user provided a custom value, fill in
+                    pluginConfig = { port: answers[`${pluginName}Port`] }
+                }
+            }
+            config.plugins![pluginName] = pluginConfig
         }
+    })
 
-        return wallet
-    }
+    config.ethereumPrivateKey = (answers.importPrivateKey) ? answers.importPrivateKey : Wallet.createRandom().privateKey
 
-    async selectPlugins(): Promise<Array<string>>{
-        const plugins: inquirer.Answers = await this.inquirerSinglePrompt( {
-            type: 'checkbox',
-            name:'selectedItems',
-            message: 'Select the plugins to enable',
-            choices: [
-                {name: 'Websocket'},
-                {name:'MQTT'},
-                {name:'HttpPublish'}
-            ]
-        })
+    return config as Config
```

Comment:
The JSON is not a `Config` type (which is the type definition for the configuration file that we get **after** we parse it by `Ajv`)